test/e2e/apps/replica_set.go

```diff
@@ -445,13 +448,23 @@ func testRSLifeCycle(f *framework.Framework) {
 	}
 
 	rsName := "test-rs"
+	label := "test-rs=patched"
+	labelMap := map[string]string{"test-rs": "patched"}
 	replicas := int32(1)
 	rsPatchReplicas := int32(3)
 	rsPatchImage := imageutils.GetE2EImage(imageutils.Pause)
 
+	w := &cache.ListWatch{
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.LabelSelector = label
+			return f.ClientSet.AppsV1().ReplicaSets(ns).Watch(context.TODO(), options)
+		},
+	}
+	rsList, err := f.ClientSet.AppsV1().Deployments("").List(context.TODO(), metav1.ListOptions{LabelSelector: label})
```

Comment:
This should list replicasets not deployments.

Reply:
oh, yes. strange how tests can pass in this case..

Reply:
fixed. thanks!

Reply:
the list was only used for the resourceversion, which would be the same as long as replicasets and deployments are using the same etcd